pages/submissionFiles/submissionFiles.sql

```diff
@@ -0,0 +1,21 @@
+-- BLOCK select_submission_file
+WITH submission_with_file AS (
+    SELECT
+        s.id,
+        jsonb_array_elements(s.submitted_answer->'_files') AS file
+    FROM
+        submissions AS s
+        JOIN variants AS v ON (v.id = s.variant_id)
+    WHERE
+        s.id = $submission_id
+        -- We check both `question_id` and `instance_question_id` to make sure that
+        -- this variant is actually accessible by the user. Both will have been
+        -- validated by middleware. For the instructor preview page, we don't have
+        -- an instance question, so we ignore it in that case.
+        AND v.question_id = $question_id
+        AND (($has_instance_question IS FALSE) OR (v.instance_question_id = $instance_question_id))
```

Comment:
Elsewhere we've used the pattern that `$instance_question_id` is passed as NULL to indicate not to check it, which would then give
```suggestion
        AND (($instance_question_id::bigint IS NULL) OR (v.instance_question_id = $instance_question_id))
```

Reply:
(but I'm also fine with the explicit boolean flag if you prefer that)

Reply:
I like that better! I copied this from #5500 which git-blames back to me 🙃

Reply:
Changed in 3aa81c1.